### PR TITLE
Remove libexecinfo on Alpine

### PR DESCRIPTION
**Pull request by morrisonlevi**
### Description

Fix GH-1824. The quick summary is Alpine 3.17 removed this library, and on older versions it's apparently not great quality anyway.

The biggest negative impact of this change is that when `DD_TRACE_HEALTH_METRICS_ENABLED` or `DD_LOG_BACKTRACE` or their INI equivalents are used, and a `SIGSEGV` occurs, that a backtrace will no longer be displayed. In other words, although this is a breaking change for Alpine platforms, the impact is quite small.

On the other hand, it has one less runtime dependency; Alpine users will no longer need to `apk add libexecinfo` before installing the tracer.

### Readiness checklist
- [x] Changelog has been added to the release document.
- [x] Tests added for this feature/bug.

### Reviewer checklist
- [x] Appropriate labels assigned.
- [x] Milestone is set.

### CODE REVIEW

**Review by realFlowControl — APPROVED** [review]
Looks good to me, but I think @bwoebi / @krakjoe should have a look before merging

**Comment by r1viollet** [comment]
can you see if the musl-dev package does not have the backtrace api ?

File: dockerfiles/verify_packages/installer/test_install_on_plesk.sh
Comment (bwoebi) [inline dockerfiles/verify_packages/installer/test_install_on_plesk.sh:16]: I'd definitely bet on it being on purpose :-) Guess we can just remove it.

**Review by bwoebi — APPROVED** [review]
LGTM :-)

**Pull request merged**
